src/Lecture2.hs

```diff
@@ -70,7 +80,15 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt index list = if index >= 0 && index < length list then (Just (list !!index), removeN index list) else (Nothing, list)
```

Comment:
Which alternative can I use? I still don't understand

Reply:
The alternative is not to use `length` at all. Here you use it to check if the index is in bounds. So, the solution that doesn't use `length` doesn't check if the index is in bounds. You can compare with zero because it's quick. But the optimal lazy solution just walks through the list until the list is empty or the index reaches the element.